app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt

```diff
@@ -90,322 +90,403 @@ class TopicPracticeFragmentTest {
 
   private var skillIdList = ArrayList<String>()
   private val internalProfileId = 0
-  private lateinit var activityScenario: ActivityScenario<TopicActivity>
 
-  @get:Rule
-  var activityTestRule: ActivityTestRule<TopicActivity> = ActivityTestRule(
-    TopicActivity::class.java,
-    /* initialTouchMode= */ true,
-    /* launchActivity= */ false
-  )
+  @Inject
+  lateinit var context: Context
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
 
   @Before
   fun setUp() {
-    FirebaseApp.initializeApp(ApplicationProvider.getApplicationContext())
-    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
-    activityScenario = launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID)
     Intents.init()
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+    FirebaseApp.initializeApp(context)
     skillIdList.add("5RM9KPfQxobH")
     skillIdList.add("B39yK4cbHZYI")
   }
 
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+    Intents.release()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
   @Test
-  // TODO(#973): Fix TopicPracticeFragmentTest
-  @Ignore
   fun testTopicPracticeFragment_loadFragment_displaySubtopics_startButtonIsInactive() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
+      testCoroutineDispatchers.runCurrent()
       onView(
         allOf(
           withText(TopicTab.getTabForPosition(2).name),
           isDescendantOfA(withId(R.id.topic_tabs_container))
         )
       ).perform(click())
-      onView(withId(R.id.master_skills_text_view))
```

Comment:
Here & below: can you provide some brief background on why these changes were needed?

Reply:
`master_skills_text_view ` is not a direct text view, it's an item in the first position of RecyclerView, so to match the text with the 0th index of an item in the RecyclerView this change is needed.

Reply:
Ah, weird that it was written in this way originally then. That makes sense--thanks!